src/mlpack/bindings/go/print_doc_functions_impl.hpp

```diff
@@ -0,0 +1,328 @@
+  // Continue recursion.
+  std::string rest = PrintOptionalInputs(args...);
+  if (rest != "" && result != "")
+    result += rest;
+  else if (result == "")
+    result = rest;
+
+  return result;
+}
+
+// Recursion base case.
+std::string PrintInputOptions() { return ""; }
+
+/**
+ * Print an input option.  This will throw an exception if the parameter does
+ * not exist in CLI.
+ */
+template<typename T, typename... Args>
+std::string PrintInputOptions(const std::string& paramName,
+                              const T& value,
+                              Args... args)
+{
+  // See if this is part of the program.
+  std::string result = "";
+
+  if (CLI::Parameters().count(paramName) > 0)
+  {
+    const util::ParamData& d = CLI::Parameters()[paramName];
+    if (d.input && d.required)
+    {
+      // Print the input option.
+      std::ostringstream oss;
+      oss << PrintValue(value, d.tname == TYPENAME(std::string));
+      result = oss.str();
+    }
+  }
+  else
+  {
+    // Unknown parameter!
+    throw std::runtime_error("Unknown parameter '" + paramName + "' " +
+        "encountered while assembling documentation!  Check PROGRAM_INFO() " +
+        "declaration.");
+  }
+
+  // Continue recursion.
+  std::string rest = PrintInputOptions(args...);
+  if (rest != "" && result != "")
+    result += ", " + rest;
+  else if (result == "")
+    result = rest;
+
+  return result;
+}
+
+// Recursion base case.
+inline std::string PrintOutputOptions() { return ""; }
+
+template<typename T, typename... Args>
+std::string PrintOutputOptions(const std::string& paramName,
+                               const T& value,
+                               Args... args)
+{
+  // See if this is part of the program.
+  std::string result = "";
+  if (CLI::Parameters().count(paramName) > 0)
+  {
+    const util::ParamData& d = CLI::Parameters()[paramName];
+    if (!d.input)
+    {
+      // Print a new line for the output option.
+      std::ostringstream oss;
+      oss << value;
+      result = oss.str();
+    }
+  }
+  else
+  {
+    // Unknown parameter!
+    throw std::runtime_error("Unknown parameter '" + paramName + "' " +
+        "encountered while assembling documentation!  Check PROGRAM_INFO() " +
+        "declaration.");
+  }
+
+  // Continue recursion.
+  std::string rest = PrintOutputOptions(args...);
+  if (rest != "" && result != "")
+    result += ", " + rest;
+  else if (result == "")
+    result = rest;
+
+  return result;
+}
+
+
+/**
+ * Given a name of a binding and a variable number of arguments (and their
+ * contents), print the corresponding function call.
+ */
+template<typename... Args>
+std::string ProgramCall(const std::string& programName, Args... args)
+{
+  std::string result = "";
+  std::string goProgramName = programName;
+  goProgramName[0] = std::toupper(goProgramName[0]);
+
+  // Initialize the method parameter structure
+  std::ostringstream oss;
+  oss << ">>> param := Initialize" << goProgramName << "()\n";
```

Comment:
The `>>>` here was meant to imitate the Python REPL, but I don't know if Go has any equivalent thing.  You might consider just having a couple spaces indentation, i.e. `"  param := Initialize"` instead, but I'm not picky on how to do it.